deepchem/models/tests/test_losses.py

```diff
@@ -381,3 +384,131 @@ def test_ShannonEntropy_pytorch(self):
             -np.mean([0.9 * np.log(0.9), 0.1 * np.log(0.1)])
         ]
         assert np.allclose(expected, result)
+
+    @pytest.mark.torch
+    def test_GlobalMutualInformation_pytorch(self):
+        """."""
+        from deepchem.feat.graph_data import BatchGraphData
+        from deepchem.models.torch_models.infograph import InfoGraphEncoder
+        from deepchem.models.torch_models.layers import MultilayerPerceptron
+        torch.manual_seed(123)
+
+        data, _ = self.get_regression_dataset()
+        batch = BatchGraphData(data.X).numpy_to_torch()
+
+        num_feat = 30
+        edge_dim = 11
+        dim = 4
+
+        encoder = InfoGraphEncoder(num_feat, edge_dim, dim)
+        encoding, feature_map = encoder(batch)
+
+        g_enc = MultilayerPerceptron(2 * dim, dim)(encoding)
+        g_enc2 = MultilayerPerceptron(2 * dim, dim)(encoding)
```

Comment:
Same comment here; let's use a tensor of the correct shape and not tie this test to InfoGraph/MultilayerPerceptron